mlflow/models/evaluation/base.py

```diff
@@ -0,0 +1,530 @@
+    :param run_id: The ID of the MLflow Run to which to log results. If
+                   unspecified, behavior depends on the specified `evaluator`.
+                   When `run_id` is unspecified, the default evaluator logs
+                   results to the current active run, creating a new active run if
+                   one does not exist.
+    :param evaluators: The name of the evaluator to use for model evaluations, or
+                       a list of evaluator names. If unspecified, all evaluators
+                       capable  of evaluating the specified model on the specified
+                       dataset are used. The default evaluator can be referred to
+                       by the name 'default'.
+    :param evaluator_config: A dictionary of additional configurations to supply
+                             to the evaluator. If multiple evaluators are
+                             specified, each configuration should be supplied as
+                             a nested dictionary whose key is the evaluator name.
+    :return: An `EvaluationResult` instance containing evaluation results.
+    """
+    # import _model_evaluation_registry and PyFuncModel inside function to avoid circuit importing
+    from mlflow.models.evaluation.evaluator_registry import _model_evaluation_registry
+    from mlflow.pyfunc import PyFuncModel
+
+    if not evaluators:
+        evaluators = "default"
+
+    if isinstance(evaluators, str):
+        evaluators = [evaluators]
+        if not (evaluator_config is None or isinstance(evaluator_config, dict)):
+            raise ValueError(
+                "If `evaluators` argument is a str, evaluator_config must be None " "or a dict."
+            )
+        evaluator_config = {evaluators[0]: evaluator_config}
+    elif isinstance(evaluators, list):
+        evaluators = set(evaluators)
+        if not (
+            isinstance(evaluator_config, dict)
+            and all(k in evaluators and isinstance(v, dict) for k, v in evaluator_config.items())
+        ):
+            raise ValueError(
+                "If `evaluators` argument is a evaluator name list, evaluator_config"
+                "must be a dict contains mapping from evaluator name to individual "
+                "evaluator config dict."
+            )
+
+    if isinstance(model, str):
+        model = mlflow.pyfunc.load_model(model)
+    elif isinstance(model, PyFuncModel):
+        pass
+    else:
+        raise ValueError(
+            "The model argument must be a URI str referring to mlflow model or "
+            "an instance of `mlflow.pyfunc.PyFuncModel`."
+        )
+
+    with StartRunOrReuseActiveRun(run_id) as actual_run_id:
+        client = mlflow.tracking.MlflowClient()
+        dataset._log_dataset_tag(client, actual_run_id)
+
+        eval_results = []
+        for evaluator_name in evaluators:
+            config = evaluator_config.get(evaluator_name)
+            if config is None:
+                config = {}
```

Comment:
```suggestion
            config = evaluator_config.get(evaluator_name, {})
```

Can we set the default value?

Reply:
we still need `if config is None: config = {}` because :

```
v = {'a': None}
v.get('a', {}) # will return None
```

Reply:
That makes sense. 

```suggestion
            config = evaluator_config.get(evaluator_name) or {}
```